### PR TITLE
Contextual header

### PR DESCRIPTION
**Pull request by bartaz**
screen shot 2017-02-21 at 15 25 18

Currently relying on `store_name` and `snapcraft_data` in snaps response which is not fully supported yet, so the states may not be properly detected, but UI is done.

### CODE REVIEW

File: src/common/components/first-time-heading/index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+const hasStoreNameButNotSnapcraftData = (snap) => snap.store_name && !snap.snapcraft_data;
+
+class FirstTimeHeading extends Component {
+
+  hasNoBuilds(snap) {
+    const { fullName } = parseGitHubRepoUrl(snap.git_repository_url);
+    const repoBuilds = this.props.snapBuilds[fullName];
+
+    if (repoBuilds && repoBuilds.success) {
+      // if builds for given repo were fetched by there aren't any
```
Comment (cjwatson) [inline src/common/components/first-time-heading/index.js]: EPARSE

File: src/common/components/repositories-home/index.js [hunk 2]
```diff
@@ -47,13 +49,16 @@ class RepositoriesHome extends Component {
...
+    const { snaps, snapBuilds } = this.props;
+
     return (
       <div>
-        <h2>Repos to build and publish</h2>
+        <FirstTimeHeading snaps={snaps} snapBuilds={snapBuilds} />
+        <HeadingThree>Repos to build and publish</HeadingThree>
         <div className={ styles['button-container'] }>
-          <Anchor appearance="neutral" href="/dashboard/select-repositories">
+          <LinkButton appearance="neutral" to="/dashboard/select-repositories/">
```
Comment (cjwatson) [inline src/common/components/repositories-home/index.js]: The route definition in `src/common/routes.js` doesn't have the trailing slash.

File: src/common/components/first-time-heading/index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    const snapsStore = this.props.snaps;
+    const snaps = snapsStore.snaps;
+
+    let message = null;
+    let progress = null;
+
+    if (snapsStore.success) {
+      // no repos added yet
+      if (snaps.length === 0) {
+        message = 'Let’s get started! First, choose one or more GitHub repos for building.';
```
Comment (katiefenn) [inline src/common/components/first-time-heading/index.js]: Might be nice to define these as constants at the top of the page.

File: src/common/components/first-time-heading/index.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        progress = [SIGNALS.DONE, SIGNALS.ACTIVE, SIGNALS.DEFAULT];
+      // at least one repo, but none have a name yet
+      } else if (snaps.filter(hasStoreName).length === 0) {
+        message = 'Great! To publish a snap to the Store, it needs a unique name. Try registering one now.';
+        progress = [SIGNALS.DONE, SIGNALS.DONE, SIGNALS.ACTIVE];
+      // at least one repo has a name but no snapcraft.yaml, and none have both
```
Comment (katiefenn) [inline src/common/components/first-time-heading/index.js]: Optional enhancement: extract these conditions into helper functions:

`hasNoRepos(snaps)`
`hasOneOrMoreRepo(snaps) && hasNoNamedRepos(snaps)`
`hasOneOrMoreNamedRepos(snaps) && hasNoDoneRepos(snaps)`

**Pull request merged**